Add a Pre-requisite: `brew install go`

On my fresh new Mac M1, I couldn't run the `install`- command without installing [`go`](https://formulae.brew.sh/formula/go) first.

Either we need to inform the user by adding this to the Pre-Requisite or it should "bin"- ready (without having Go installed) if you know what I mean.

## Expected Behavior

Run

```sh
m1-terraform-provider-helper install hashicorp/template -v v2.2.0
```

Normal success

## Actual Behavior

```sh
m1-terraform-provider-helper install hashicorp/template -v v2.2.0
Repo: https://github.com/hashicorp/terraform-provider-template
GitRepo: git@github.com:hashicorp/terraform-provider-template
2022/01/19 11:31:22 Executingcd /Users/johnnguyen/.m1-terraform-provider-helper && git clone git@github.com:hashicorp/terraform-provider-template
Cloning into 'terraform-provider-template'...
Cancel
2022/01/19 11:31:25 Cloning into 'terraform-provider-template'...

HEAD is now at 477a7b5 Cleanup after v2.2.0 release
2022/01/19 11:31:31 version: v2.2.0
==> Checking that code complies with gofmt requirements...
/Users/foo/.m1-terraform-provider-helper/terraform-provider-template/scripts/gofmtcheck.sh: line 5: gofmt: command not found
go install
make: go: No such file or directory
make: *** [build] Error 1
2022/01/19 11:31:32 Bash code did not run successfully: exit status 2
```


## Your Environment

- OS: MacOs Monterey (12.1)
- m1-terraform-provider-helper version: 0.3.1

--- a/.goreleaser.yml
+++ b/.goreleaser.yml
@@ -25,12 +25,12 @@ changelog:
   sort: asc
 
 brews:
-  -
-    tap:
+  - tap:
       owner: kreuzwerker
       name: homebrew-taps
       token: "{{ .Env.GORELEASER_TOKEN }}"
-
+    dependencies:
+      - name: go
     download_strategy: CurlDownloadStrategy
     homepage: "https://github.com/kreuzwerker/m1-terraform-provider-helper"
     description: "CLI to support with downloading and compiling terraform providers for Mac with M1 chip."

